File: greedy/jump-game/jump-game.java
class Solution {
  public boolean canJump(int[] nums) {
    int n = nums.length;
    int jump = 0;

    for (int i = 0; i < n - 1; i++) {
      jump = Math.max(jump - 1, nums[i]);

      if (jump == 0) {
        return false;
      }
    }

    return true;
  }
}

File: greedy/jump-game/jump-game.go
func canJump(nums []int) bool {
	n := len(nums)
	jump := 0

	for i := 0; i < n-1; i++ {
		if jump-1 < nums[i] {
			jump = nums[i]
		} else {
			jump--
		}

		if jump == 0 {
			return false
		}
	}

	return true
}
